packages/web/pages/api/pool/[id].ts

```diff
@@ -6,8 +6,8 @@ type Response = {
 
 export default async function pools(req: Request) {
   const url = new URL(req.url);
-  const poolIdParam = url.searchParams.has("id")
-    ? url.searchParams.get("id")
+  const poolIdParam = url.pathname.split("/").slice(-1)
+    ? url.pathname.split("/").slice(-1)
     : undefined;
```

Comment:
q: why does it have to be a conditional and not initialized directly?

Reply:
Indeed, that piece of code was redundant. I have uploaded an updated version with enhanced validation behavior.